manifests/plugin/salt.pp

```diff
@@ -0,0 +1,47 @@
+# = Foreman Proxy Salt plugin
+#
+# This class installs Salt plugin
+#
+# === Parameters:
+#
+# $autosign_file::       File to use for salt autosign
+#
+# $enabled::             Enables/disables the plugin
+#                        type:boolean
+#
+# $listen_on::           Proxy feature listens on https, http, or both
+#
+# $group::               Owner of plugin configuration
+#
+# $salt_cmd::            Location of salt binary
+#
+# $salt_key_cmd::        Location of salt-key binary
+#
+# $salt_command_user::   User to run salt commands under
+#
+class foreman_proxy::plugin::salt (
+  $autosign_file     = $::foreman_proxy::plugin::salt::params::autosign_file,
+  $enabled           = $::foreman_proxy::plugin::salt::params::enabled,
+  $listen_on         = $::foreman_proxy::plugin::salt::params::listen_on,
+  $group             = $::foreman_proxy::plugin::salt::params::group,
+  $salt_cmd          = $::foreman_proxy::plugin::salt::params::salt_cmd,
+  $salt_key_cmd      = $::foreman_proxy::plugin::salt::params::salt_key_cmd,
+  $salt_command_user = $::foreman_proxy::plugin::salt::params::salt_command_user,
+) inherits foreman_proxy::plugin::salt::params {
+
+  $group_real = pick($group, $::foreman_proxy::user)
+
+  validate_absolute_path($autosign_file, $salt_cmd, $salt_key_cmd)
+  validate_bool($enabled)
+  validate_string($group_real)
+
+  foreman_proxy::plugin {'salt':
+  } ->
+  file {'/etc/foreman-proxy/settings.d/salt.yml':
```

Comment:
We have ```foreman_proxy::settings_file``` for this. We converted the other plugins already so have a look at those.